shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -136,8 +133,9 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, val ne
       * e.g. `code"x => y => x + y"`. For `y => x + y`, freeVarsInCurrentQuote = {'gx, 'gy}, quoteSkolemEnv = {'gy}.
       * To get the contextual type, we solve constraint the 'gx \/ 'gy <= 'a \/ 'gy, where 'a is a fresh type variable for `y => x + y`'s contextual type.
       * So for `code"x => ..."`, freeVarsInCurrentQuote = {'a}, quoteSkolemEnv = {'gx}, where 'gx <= 'a.
+      * After calling `enterQuotedScope`, **solve the constraints** using `solveQuoteContext` to make sure free variables are handled correctly.
       */
-    def enterQuotedScope: Ctx = copy(Some(this), MutMap.empty, MutMap.empty, inQuote = true, quoteSkolemEnv = MutMap.empty, freeVarsInCurrentQuote = MutSet.empty)
+    def enterQuotedScope: Ctx = copy(Some(this), MutMap.empty, MutMap.empty, lvl = lvl + 1, inQuote = true, quoteSkolemEnv = MutMap.empty, freeVarsInCurrentQuote = MutSet.empty)
```

Comment:
This is wrong. Why did you suddenly decide to change the level here? It's a _very_ significant (and unnecessary) change in semantics.

Reply:
Because we need to make sure the skolem tags have higher levels🤔.

Reply:
Why did you _just_ make the change an hour ago, then? How was it dealt with before?

Reply:
Before that, a `poly` always follows an `enterQuotedScope`. That's why we only check `ctx.inQuote` in the `doGenLamda` case in the previous versions.

Reply:
Oh my, so THAT'S why you only wanted to use that case 🤦 

Ok, then all good now I think.

You should really learn to better synthesize the reasons you made things the way they are AND then self-reflect critically about whether that's really the optimal way. Because the previous design made very little sense. Thankfully I caught it in the review process.